vignettes/compareToBaseR.Rmd

```diff
@@ -0,0 +1,683 @@
+
+```{r}
+## base
+nchar(letters)
+
+## stringr
+str_length(letters)
+
+```
+
+There are some subtle differences between base and stringr here. `nchar()` requires a character vector, so it will return an error if used on a factor. `str_length()` can handle a factor input.
+
+
+```{r,eval=F}
+## base
+nchar(factor("abc")) ## Error in nchar(factor("abc")) : 'nchar()' requires a character vector
+```
+
+```{r}
+## stringr
+str_length(factor("abc"))
+```
+
+From the documentation: "Technically this [`str_length()`] returns the number of 'code points', in a string. One code point usually corresponds to one character, but not always."
+
+```{r}
+u1 <- "\u00fc"
+u2 <- stringi::stri_trans_nfd(u1)
+
+u1
+u2
+
+## base
+nchar(u1) 
+nchar(u2)
+
+
+## stringr
+str_length(u1)
+str_length(u2)
+
+```
+
+## `str_pad`
+
+*Pad a string*
+
+If you want to pad a string to a certain width, you can use stringr's `str_pad()`. In base R you could use `sprintf()`, but unlike `str_pad()`, `sprintf()` has many other functionalities. 
+
+```{r}
+## stringr
+rbind(
+  str_pad("hadley", 30, "left"),
+  str_pad("hadley", 30, "right"),
+  str_pad("hadley", 30, "both")
+)
+
+## base
+sprintf("%30s", "hadley")
+sprintf("%-30s", "hadley")
+# "both" is not as straightforward
+```
+
+
+## `str_trunc`
+
+*Truncate a character string*
+
+The stringr package provides an easy way to truncate a character string: `str_trunc()`. In base R you would have to do something a bit more complicated, potentially leveraging `substr()`.
+
+```{r}
+x <- "This string is moderately long"
+
+## stringr
+rbind(
+  str_trunc(x, 20, "right"),
+  str_trunc(x, 20, "left"),
+  str_trunc(x, 20, "center")
+)
+
+```
+
+## `str_trim`
+
+*Trim whitespace from a string*
+
+Similarly, stringr provides `str_trim()` to trim whitespace from a string. This is analogous to base R's `trimws()`.
+
+```{r}
+## stringr
+str_trim(" String with trailing and leading white space\t")
+str_trim("\n\nString with trailing and leading white space\n\n")
+
+## base
+trimws(" String with trailing and leading white space\t")
+trimws("\n\nString with trailing and leading white space\n\n")
+```
+
+The stringr function `str_squish()` allows for extra whitespace within a string to be trimmed (in contrast to `str_trim()` which removes whitespace at the beginning and/or end of string). In base R, one might take advantage of `gsub()` to accomplish the same effect.
+
+```{r}
+## stringr
+str_squish(" String with trailing, middle,   and leading white space\t")
+str_squish("\n\nString with excess, trailing and leading white space\n\n")
+```
+
+## `str_wrap`
+
+*Wrap strings into nicely formatted paragraphs*
+
+I'm not sure why these outputs don't match. The stringr packages's `str_wrap()` is a wrapper around `stringi::stri_wrap()` which has defaults that are supposed to be compatible with base R's `strwrap()`.
```

Comment:
I don't think the implementations are supposed to match — stringi uses Knuth-Plauger